Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -505,18 +505,18 @@ instance [PartialOrder α] : Subsingleton (SuccOrder α) :=
     · exact (@IsMax.succ_eq _ _ h₀ _ ha).trans ha.succ_eq.symm
     · exact @CovBy.succ_eq _ _ h₀ _ _ (covBy_succ_of_not_isMax ha)⟩
 
-section CompleteLattice
-
-variable [CompleteLattice α] [SuccOrder α]
-
-theorem succ_eq_iInf (a : α) : succ a = ⨅ (b) (_ : a < b), b := by
```

Comment:
Can you keep `succ_eq_iInf` around?

Reply:
Wait, we don't seem to have the conditionally complete version of that? That's probably what we should be fixing instead.

Reply:
> Wait, we don't seem to have the conditionally complete version of that? That's probably what we should be fixing instead.

Yeah, because it's false. Nested conditional `iInf`s don't behave the way you expect.

Reply:
`succ_eq_ciInf` is not true in general, and requires that the infimum be less than `sInf Ø`. Here's what the new proof of `succ_eq_iInf` looks like:
```
theorem succ_eq_iInf [CompleteLattice α] [SuccOrder α] (a : α) : succ a = ⨅ (b) (_ : a < b), b := by
  rw [succ_eq_sInf, iInf_subtype', iInf, Subtype.range_coe_subtype]
  rfl
```
Again, I feel like it's a bit dumb to have this, and if we will, why not `⨅ b ∈ Set.Iio a, b` instead? I'll add it if you really want it, but I personally don't get it.

Reply:
I mean, consider lemmas like `le_iInf₂`. `succ_eq_sInf` would be annoying to interface with it.